forecasting/src/autogluon/forecasting/task/forecasting/forecasting.py

```diff
@@ -0,0 +1,73 @@
+from core.task.base.base_task import BaseTask, schedulers, compile_scheduler_options
+from .dataset import TimeSeriesDataset
+
+from ...learner.abstract_learner import AbstractLearner
+from ...learner import DefaultLearner as Learner
+from .predictor import ForecastingPredictor
+from ...trainer import AutoTrainer
+from core.utils.utils import setup_outputdir
+__all__ = ['Forecasting']
+
+
+class Forecasting(BaseTask):
+
+    Dataset = TimeSeriesDataset
+    Predictor = ForecastingPredictor
+
+    @staticmethod
+    def load(output_directory, verbosity=2):
+        return ForecastingPredictor.load(output_directory=output_directory, verbosity=verbosity)
+
+    @staticmethod
+    def fit(train_data,
+            freq,
```

Comment:
Why do we need `freq` here if it can be auto-inferred?

Reply:
my understanding here is that the user first use the class Timeseriesdataset to transform the data outside the task to get freq and prediction length, which should be auto-inferred, and then use them to put into the task. Should we put this step inside the task.fit()?